lib/core/versions/latest/BatchWriter.ts

```diff
@@ -49,6 +52,14 @@ export default class BatchWriter implements IBatchWriter {
       return 0;
     }
 
+    const lastSubmitted = await this.confirmationStore.getLastSubmitted();
+    if (lastSubmitted !== null &&
+      (lastSubmitted.confirmedAt === undefined || currentTime.time - lastSubmitted.confirmedAt < this.minConfirmationBetweenWrites - 1)
```

Comment:
Why the -1 at the end?

Reply:
Say comfirmedAt = 100, currentTime = 105, then the message have already been confirmed 6 times in block 100 to 105. Hence it's okay to proceed. It only needs to wait if currentTime - confirmedAt < 5. What do you think?